From 7659feceda1c645d1931fa6a34ec6ea7075dccce Mon Sep 17 00:00:00 2001
From: Michael Shych <michaelsh@nvidia.com>
Date: Thu, 23 Mar 2023 17:59:25 +0000
Subject: =?UTF-8?q?From=20c93c44e9ffb798072a22557d0be7a9290a5affda=20Mon?=
 =?UTF-8?q?=20Sep=2017=2000:00:00=202001=0ASubject:=20[PATCH=20backport=20?=
 =?UTF-8?q?6.1.42=2084/85]=20e1000e:=20OPT:=20skip=20NVM=20checksum.?=

NVM checksum is reported as bad in many cases after full
BIOS SPI flash including GbE section burn by Dediprog.
This happened in the early steps of production flow before
EEPROM / NVM info will be customized and the checksum will
be recalculated. Just skip this check.

Signed-off-by: Michael Shych <michaelsh@nvidia.com>
---
 drivers/net/ethernet/intel/e1000e/netdev.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/net/ethernet/intel/e1000e/netdev.c b/drivers/net/ethernet/intel/e1000e/netdev.c
index cbd8357c6..8940b41de 100644
--- a/drivers/net/ethernet/intel/e1000e/netdev.c
+++ b/drivers/net/ethernet/intel/e1000e/netdev.c
@@ -7568,6 +7568,12 @@ static int e1000_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
 	/* systems with ASPM and others may see the checksum fail on the first
 	 * attempt. Let's give it a few tries
 	 */
+	/* NVIDIA >>>
+	 * NVM checksum is reported as bad in many cases after full
+	 * BIOS SPI flash including GbE section burn by Dediprog.
+	 * This happened in the early steps of production flow before
+	 * EEPROM / NVM info will be customized and the checksum will
+	 * be recalculated. Just skip this check.
 	for (i = 0;; i++) {
 		if (e1000_validate_nvm_checksum(&adapter->hw) >= 0)
 			break;
@@ -7577,6 +7583,8 @@ static int e1000_probe(struct pci_dev *pdev, const struct pci_device_id *ent)
 			goto err_eeprom;
 		}
 	}
+	 * NVIDIA <<<
+	 */
 
 	e1000_eeprom_checks(adapter);
 
-- 
2.44.0

